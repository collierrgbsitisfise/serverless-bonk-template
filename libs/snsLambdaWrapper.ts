import { SNSEvent, Context, Callback } from 'aws-lambda';

export const snsLambdaWrapper = (
  lambda: (event: SNSEvent, context: Context, callback: Callback) => Promise<any>,
  onSucces: (event: SNSEvent, result: any) => any | PromiseLike<any>,
  onError: (event: SNSEvent, error: Error) => any | PromiseLike<any>,
) => {
  return function wrapp(event: SNSEvent, context?: Context, callback?: Callback): Promise<any> {
    return Promise.resolve()
      .then(() => lambda(event, context, callback))
      .then((res: any) => onSucces(event, res))
      .catch((err: Error) => onError(event, err));
  };
};
