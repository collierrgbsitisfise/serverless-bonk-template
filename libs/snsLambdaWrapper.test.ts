import { SNSEvent, Context, Callback } from 'aws-lambda';

import { snsLambdaWrapper } from './snsLambdaWrapper';

describe('snsLambdaWrapper', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });

  const fakeLambdaResponse = {
    data: 'some data',
  };
  const fakeEvent = {} as SNSEvent;
  const fakeContext = {} as Context;
  const fakeCallBack = jest.fn() as Callback;

  it('should call "onSucces" handler', async () => {
    const lambda = jest.fn().mockReturnValueOnce(fakeLambdaResponse);
    const successHandler = jest.fn();
    const errorHandler = jest.fn();

    const executable = snsLambdaWrapper(lambda, successHandler, errorHandler);
    expect.assertions(4);
    const res = await executable(fakeEvent, fakeContext, fakeCallBack);
    expect(res).toMatchSnapshot();
    expect(lambda).toBeCalledWith(fakeEvent, fakeContext, fakeCallBack);
    expect(successHandler).toBeCalledWith(fakeEvent, fakeLambdaResponse);
    expect(errorHandler).not.toBeCalledWith();
  });

  it('should call "onError" handler', async () => {
    const lambda = jest.fn().mockRejectedValueOnce(fakeLambdaResponse);
    const successHandler = jest.fn();
    const errorHandler = jest.fn();

    const executable = snsLambdaWrapper(lambda, successHandler, errorHandler);
    expect.assertions(4);
    const res = await executable(fakeEvent, fakeContext, fakeCallBack);
    expect(res).toMatchSnapshot();
    expect(lambda).toBeCalledWith(fakeEvent, fakeContext, fakeCallBack);
    expect(successHandler).not.toBeCalledWith();
    expect(errorHandler).toBeCalledWith(fakeEvent, fakeLambdaResponse);
  });
});