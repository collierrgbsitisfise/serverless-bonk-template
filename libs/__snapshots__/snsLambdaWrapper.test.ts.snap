// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snsLambdaWrapper should call "onError" handler 1`] = `undefined`;

exports[`snsLambdaWrapper should call "onSucces" handler 1`] = `undefined`;
